### PR TITLE
[Relay] GetItem

### PR DESCRIPTION
**Pull request by MarisaKirisame**
GetItem is a relay construct for getting field out of a tuple

### CODE REVIEW

**Comment by MarisaKirisame** [comment]
@junrushao1994 @srkreddy1238 @masahi please review.

File: include/tvm/relay/expr.h [hunk 1]
```diff
@@ -378,6 +376,23 @@ class IfNode : public ExprNode {
...
 RELAY_DEFINE_NODE_REF(If, IfNode, Expr);
 
+/*! \brief Get a field out of a tuple. */
+class GetItem;
+class GetItemNode : public ExprNode {
+ public:
+  /*! \brief The tuple */
+  Expr tuple;
+  /*! \brief which value to get */
+  size_t field;
```
Comment (tqchen) [inline include/tvm/relay/expr.h]: change to ```int index;```

Comment (tqchen) [reply include/tvm/relay/expr.h]: the main reason is that usually if we want to serialize the model, the type should have fixed size across platform(int32, or int64). index is a more commonly used term

File: include/tvm/relay/expr.h [hunk 2]
```diff
@@ -378,6 +376,23 @@ class IfNode : public ExprNode {
 
 RELAY_DEFINE_NODE_REF(If, IfNode, Expr);
 
+/*! \brief Get a field out of a tuple. */
+class GetItem;
```
Comment (tqchen) [inline include/tvm/relay/expr.h]: GetItem->TupleGetItem

**Comment by MarisaKirisame** [comment]
@junrushao1994 thx for reviewing!
@jroesch the relay doc/evaluator should be updated with tuplegetitem
@joshpoll the relay prettyprinter/parser/text format should be updated with tuplegetitem
I can take the responsibility of adding tuplegetitem if you two are busy - in that case I can make pr to you two's branch.

**Pull request merged**